src/actions/tree.js

```diff
@@ -21,6 +21,10 @@ export const tree = new Baobab({
   //
   repos: {},
 
+  // the secrets branch of the tree maintains the current secrets for the repository owner/name
+
+  secrets: {},
+
```

Comment:
when the repository is deleted the secrets should be cleared. This can be done by adding this line:

```diff
      tree.unset(['repos', owner, name]);
      tree.unset(['builds', owner, name]);
+     tree.unset(['secrets', owner, name]);

```